src/01-language/coercion.ml

```diff
@@ -7,7 +7,10 @@ and ty_coercion' =
   | ArrowCoercion of ty_coercion * dirty_coercion
   | HandlerCoercion of dirty_coercion * dirty_coercion
   | TyCoercionVar of Type.TyCoercionParam.t
-  | ApplyCoercion of TyName.t * ty_coercion list
+  | ApplyCoercion of {
+      ty_name : TyName.t;
+      tcoers : (ty_coercion * variance) TyParam.Map.t;
```

Comment:
Add a TODO in the code to move the variance into `ty_constructor` when we expand `ty_name` with additional information.